api/src/main/java/com/velocitypowered/api/command/Command.java

```diff
@@ -4,10 +4,35 @@
 import com.velocitypowered.api.proxy.Player;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import org.checkerframework.checker.nullness.qual.NonNull;
 
 /**
  * Represents a command that can be executed by a {@link CommandSource}, such as
  * a {@link Player} or the console.
+ *
+ * <p>Velocity 1.1.0 introduces specialized command subinterfaces to separate
+ * command parsing concerns. These include, in order of preference:
+ *
+ * <ul>
+ * <li>{@link BrigadierCommand}, which supports parameterized arguments and
+ * specialized execution, tab complete suggestions and permission-checking logic.
+ *
+ * <li>{@link LegacyCommand}, modelled after the convention popularized by
+ * Bukkit and BungeeCord. Older classes directly implementing {@link Command}
+ * are suggested to migrate to this interface.
+ *
+ * <li>{@link RawCommand}, useful for bolting on external command frameworks
+ * to Velocity.
+ *
+ * </ul>
+ *
```

Comment:
`<p>For this reason, the legacy `execute()`, `suggest()`, and `hasPermission()` methods are deprecated and will be removed in Velocity 2.0.0. We suggest implementing one of the more specific subinterfaces instead.`